config/config.go

```diff
@@ -1,267 +1,104 @@
-		Enabled:                 EnableWebsocket,
-		Port:                    WebsocketPort,
-		MaxWriteResponseRetries: 3,
-		WriteResponseTimeout:    10 * time.Second,
-	},
-	Performance: struct {
-		WatchChanBufSize       int           `mapstructure:"watchchanbufsize"`
-		ShardCronFrequency     time.Duration `mapstructure:"shardcronfrequency"`
-		MultiplexerPollTimeout time.Duration `mapstructure:"servermultiplexerpolltimeout"`
-		MaxClients             int32         `mapstructure:"maxclients"`
-		EnableMultiThreading   bool          `mapstructure:"enablemultithreading"`
-		StoreMapInitSize       int           `mapstructure:"storemapinitsize"`
-		AdhocReqChanBufSize    int           `mapstructure:"adhocreqchanbufsize"`
-		EnableProfiling        bool          `mapstructure:"profiling"`
-	}{
-		WatchChanBufSize:       20000,
-		ShardCronFrequency:     1 * time.Second,
-		MultiplexerPollTimeout: 100 * time.Millisecond,
-		MaxClients:             int32(20000),
-		EnableMultiThreading:   false,
-		StoreMapInitSize:       1024000,
-		AdhocReqChanBufSize:    20, // assuming we wouldn't have more than 20 adhoc requests being sent at a time.
-	},
-	Memory: struct {
-		MaxMemory      int64   `mapstructure:"maxmemory"`
-		EvictionPolicy string  `mapstructure:"evictionpolicy"`
-		EvictionRatio  float64 `mapstructure:"evictionratio"`
-		KeysLimit      int     `mapstructure:"keyslimit"`
-		LFULogFactor   int     `mapstructure:"lfulogfactor"`
-	}{
-		MaxMemory:      0,
-		EvictionPolicy: EvictAllKeysLFU,
-		EvictionRatio:  0.9,
-		KeysLimit:      DefaultKeysLimit,
-		LFULogFactor:   10,
-	},
-	Persistence: struct {
-		AOFFile            string `mapstructure:"aoffile"`
-		PersistenceEnabled bool   `mapstructure:"persistenceenabled"`
-		WriteAOFOnCleanup  bool   `mapstructure:"writeaofoncleanup"`
-	}{
-		PersistenceEnabled: true,
-		AOFFile:            "./dice-master.aof",
-		WriteAOFOnCleanup:  false,
-	},
-	Logging: struct {
-		LogLevel        string `mapstructure:"loglevel"`
-		PrettyPrintLogs bool   `mapstructure:"prettyprintlogs"`
-	}{
-		LogLevel:        "info",
-		PrettyPrintLogs: true,
-	},
-	Auth: struct {
-		UserName string `mapstructure:"username"`
-		Password string `mapstructure:"password"`
-	}{
-		UserName: "dice",
-		Password: RequirePass,
-	},
-	Network: struct {
-		IOBufferLength    int `mapstructure:"iobufferlength"`
-		IOBufferLengthMAX int `mapstructure:"iobufferlengthmax"`
-	}{
-		IOBufferLength:    512,
-		IOBufferLengthMAX: 50 * 1024,
-	},
+	NumShards int `config:"num_shards" default:"-1"`
 }
 
-var defaultConfig Config
-
 func init() {
-	config := baseConfig
-	config.Logging.PrettyPrintLogs = false
-	config.Logging.LogLevel = "info"
-	defaultConfig = config
+	parser := NewConfigParser()
+	configFilePath := filepath.Join(".", DefaultConfigName)
+	if err := parser.ParseFromFile(configFilePath); err != nil {
+		slog.Warn("Failed to parse config file", slog.String("error", err.Error()), slog.String("message", "Loading default configurations"))
```

Comment:
This could ideally check which error was thrown and give more contextual logs.
1. If file was not found: Config file not found, loading default configuration.
2. If file was found but could not be parsed: Failed to parse config file, loading default configuration.